README.md

```diff
@@ -8,44 +8,47 @@ This repo offers guidance on securing Azure Storage account through a private en
+* [Azure Command Line](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli).
+* Visual Studio Code installed on your development machine. For more information, see [Download Visual Studio Code](https://code.visualstudio.com/download).
+* [OpenSSL Command Line Tool](https://github.com/openssl/openssl?tab=readme-ov-file)
 * Bash terminal, on Windows you can use WSL
-* .NET SDK 8 (if you wish to run the client and server samples for file upload)
+* [.NET SDK 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) (if you wish to run the client and server samples for file upload)
 * DNS provider if you wish to test HTTPS with your custom domain, and SSL Certificate for your custom domain
 
 > [!TIP]
 > Without a custom DNS and an SSL certificate available, you can still test the entire setup but the IoT client device portion of this quickstart will not be completed.
 
 ### Quickstart
 
+<img alt="Quickstart components diagram" src="docs/assets/quickstart-iothub-file-upload-private-link.png" width="500" />
+
 This quickstart provides instructions on how to set up Azure IoT Hub's file upload functionality through an Azure Storage account that only allows private connections, through step-by-step scripts. The scripts deploy and configure a simplified set of Azure resources to showcase Azure Storage with private endpoint, Application Gateway and Azure IoT Hub routing to Azure Storage. The Storage account disables any public Internet access and only accessible within the Virtual Network. In this quickstart there is no Azure Firewall for traffic inspection.
-Finally, two sample .NET apps interact with the resources deployed to showcase the end to end data flows.
+Finally, two sample .NET apps interact with the resources deployed to showcase the end-to-end data flows.
 
 > [!WARNING]
-> This sample currently generates self-signed certificate and key files on disk. This should be improved to leverage Azure Key Vault for securely storing and retrieving the password, certificate and key. Application Gateway supports retrieving an SSL certificate from a Key Vault account.
+> This sample currently generates self-signed certificate and key files on disk and stores these on Azure Key Vault. In a production environment usage of self-signed certificates is not an option, though we would still recommend leveraging Azure Key Vault for storing the secrets.
 
 #### Setup
 
-* Open a Bash terminal
-* `git clone` this repo
-* cd [repository name]
-* Open the project in Visual Studio Code `code .` and open a Bash terminal.
-* Log into your Azure account `az login [--tenant xxxx-xxx]` and set the default subscription `az account set -s <subscription>`
-* Prepare required environment variables to run the deployment scripts. We create a file in the folder `./temp` which is excluded from Git.
+1. Open a Bash terminal.
+2. `git clone` this repo.
+3. Move into the cloned repo folder `cd [repository name]`.
```

Comment:
~~~suggestion
1. Clone this repository and go to the root directory.

   ```bash
   git clone https://github.com/Azure-Samples/azure-edge-extensions-iothub-fileupload-privatelink
  cd azure-edge-extensions-iothub-fileupload-privatelink
  ```
~~~